pallets/zk-snarks/src/lib.rs

```diff
@@ -156,77 +156,23 @@ pub mod pallet {
 			pub_input: Vec<u8>,
 			vec_vk: Vec<u8>,
 		) -> DispatchResult {
-			// Setting the public input data.
-			let public_inputs: PublicInputsDef<T> =
-				pub_input.try_into().map_err(|_| Error::<T>::TooLongPublicInputs)?;
-			let deserialized_public_inputs = deserialize_public_inputs(public_inputs.as_slice())
-				.map_err(|_| Error::<T>::MalformedPublicInputs)?;
-			PublicInputStorage::<T>::put(public_inputs);
-			let vk: VerificationKeyDef<T> =
-				vec_vk.try_into().map_err(|_| Error::<T>::TooLongVerificationKey)?;
-			let deserialized_vk = VKey::from_json_u8_slice(vk.as_slice())
-				.map_err(|_| Error::<T>::MalformedVerificationKey)?;
-			ensure!(
-				deserialized_vk.curve == SUPPORTED_CURVE.as_bytes(),
-				Error::<T>::NotSupportedCurve
-			);
-			ensure!(
-				deserialized_vk.protocol == SUPPORTED_PROTOCOL.as_bytes(),
-				Error::<T>::NotSupportedProtocol
-			);
-			ensure!(
-				deserialized_vk.public_inputs_len == deserialized_public_inputs.len() as u8,
-				Error::<T>::PublicInputsMismatch
-			);
-			VerificationKeyStorage::<T>::put(vk);
+			let inputs = store_public_inputs::<T>(pub_input)?;
+			let vk = store_verification_key::<T>(vec_vk)?;
+			ensure!(vk.public_inputs_len == inputs.len() as u8, Error::<T>::PublicInputsMismatch);
```

Comment:
now it looks like your are ensuring public inputs match after storing verification key, imho previous version was cleaner. 

is the storage operation reverted ?

Reply:
I run a test and the input data were not stored